### PR TITLE
Dndkit

### PR DESCRIPTION
**Pull request by thomasplevy**
## Description

Replaces the use of [react-sortable-hoc](https://github.com/clauderic/react-sortable-hoc) with [@dndkit](https://github.com/clauderic/dnd-kit)

@dndkit is *in beta* but it seems quite stable.

The motivation for making this change is that `react-sortable-hoc` relies on `findDomNode` which has been deprecated from React. It's unwise for us to continue relying on this for that reason alone. Furthermore, the dev(s) behind react-sortable-hoc have stated that there will only be security and bug fixes for the lib and there aren't any plans to reimplement without the use of `findDomNode`.

In addition to replacing the library it makes various improvements (mostly with regards to readability of code) in the affected areas, which are:

+ The instructors sidebar plugin (seen on course and membership post types)
+ Options management lists (seen on forms when using dropdown, checkbox, or radio fields)

Various visual improvement have been made.

I see *one* minor issue here that I cannot figure out how to resolve and I've decided we should ship as is unless you can see how to fix it @eri-trabiccolo:

1. Create a select/radio/option field
2. View the options
3. Add a new option
4. Drag it into a new position

The transition animation doesn't work until after it's been dropped. If it's dragged again after that it'll work as expected. I can't figure out why this is happening and it doesn't happen with instructors (which uses the same base component structure).

## How has this been tested?

+ Manually
+ I should add E2E tests but I haven't done so yet

## Types of changes

+ Updates and improvements
+ I don't consider any of the deleted code (or the removal of this library) to be backwards incompatible, although maybe some of it technically is). The component functions that have been removed, renamed, modifed, etc... couldn't be accessed by any 3rd parties, only used internally, so I don't believe there's anything to worry about here. The technically backwards incompatible change is the removal of `react-sortable-hoc`. It's conceivable that some 3rd party is relying on this... This said, I don't have a problem including this in a 3.x release even though we released 2.0 this week. Thoughts @eri-trabiccolo ?

## Checklist:

- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

File: src/js/blocks/form-fields/inspect-field-options.js [hunk 1]
```diff
@@ -300,10 +300,7 @@ export default class InspectorFieldOptions extends Component {
 
 		// Pass back to the block (and remove the ID so it isn't stored.
 		setAttributes( {
-			options: options.map( ( opt ) => ( {
-				...opt,
-				id: undefined,
-			} ) ),
+			options: options.map( ( { id, ...rest } ) => rest )
```
Comment (thomasplevy) [inline src/js/blocks/form-fields/inspect-field-options.js:303]: @eri-trabiccolo check this out:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#rest_in_object_destructuring

File: src/js/blocks/form-fields/editor.scss [hunk 2]
```diff
@@ -122,6 +141,7 @@ li.llms-field-option {
 	.llms-field-opt-default-wrap,
 	.llms-del-field-opt-wrap {
 		flex: 1;
+		height: fit-content; // Make sure tooltip isn't out of place when keys are displaying.
```
Comment (eri-trabiccolo) [inline src/js/blocks/form-fields/editor.scss:142]: didn't know this `fit-content`.

File: src/js/components/sortable-list/index.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+ * @param {Object}        props
+ * @param {Object}        props.ListItem      Component used to display each item in the sortable list.
+ * @param {Object}        props.manageState   An object used to manage the state of the items in the list. Should contain 4 keys referencing functions: createItem, deleteItem, updateItem, updateItems.
+ * @param {boolean}       props.dragHandle    If `true`, flags that the `ListItem` component will provide it's own drag handle and no innate drag functionality is provided.
+ * @param {string}        props.itemClassName CSS classname applied to each ListItem.
+ * @param {Object}        props.extraProps    Extra properties passed on to ListItem.
+ * @param {string|number} props.id          Item ID.
+ * @param {number}        props.index       Index of the item within the items list.
+ * @param {Object}        props.item        Current item object.
+ * @param {boolean}       props.isDragging  If `true`, the item is being dragged, useful for adding CSS for a different look during a drag event.
```
Comment (eri-trabiccolo) [inline src/js/components/sortable-list/index.js]:
```suggestion
 * @param {string|number} props.id            Item ID.
 * @param {number}        props.index         Index of the item within the items list.
 * @param {Object}        props.item          Current item object.
 * @param {boolean}       props.isDragging    If `true`, the item is being dragged, useful for adding CSS for a different look during a drag event.
```

File: src/js/sidebar/instructors/search.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+// Internal Deps.
+import { SearchUser } from '../../components';
+
+/**
+ * User search component for searching instructors.
+ *
+ * @since [version]
+ *
+ * @param {Object}   props
```
Comment (eri-trabiccolo) [inline src/js/sidebar/instructors/search.js:19]:
```suggestion
```
is this intended for some doc rule I'm not aware about?

File: src/js/blocks/form-fields/editor.scss [hunk 5]
```diff
@@ -114,6 +120,19 @@ li.llms-field-option {
...
+			color: #5a5a5a;
+		}
+		.components-text-control__input {
+			background: #f5f5f5;
+			font-family: monospace;
+		}
+	}
+
 	.llms-drag-handle {
 		flex: 0.8;
```
Comment (eri-trabiccolo) [inline src/js/blocks/form-fields/editor.scss:134]:
```suggestion
		cursor: grab;
		margin-top: 3px;
		flex: 0.8;
```
or `cursor: move;`

File: src/js/blocks/form-fields/editor.scss [hunk 6]
```diff
@@ -101,6 +103,10 @@ li.llms-field-option {
 		z-index: 999;
 	}
 
+	.llms-drag-handle {
+		margin-top: 3px;
+	}
+
```
Comment (eri-trabiccolo) [inline src/js/blocks/form-fields/editor.scss]:
```suggestion
```
moved below, we already had a `.llms-drag-handle` definition block.

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]
Works fine, minor changes requested (css)

**Comment by eri-trabiccolo** [comment]
@thomasplevy mmm well if we decide that this should be released as 3.0, does this also mean that LifterLMS that will include this should be released as 6.0? I mean, I think that if we worry about 3rd parties that are possibly relying on that, it's conceivable that they're relying on the blocks bundled in the core plugin rather than on this. Maybe I'm wrong btw.
I don't think we should release this as 3.0 though :D

File: src/js/blocks/form-fields/inspect-field-options.js (hunk 1 above)
Comment (eri-trabiccolo) [reply src/js/blocks/form-fields/inspect-field-options.js:303]: This is very nice and fluffy and fills my heart of love.

File: src/js/blocks/form-fields/editor.scss (hunk 2 above)
Comment (thomasplevy) [reply src/js/blocks/form-fields/editor.scss:142]: It's not widely supported (firefox doesn't support it for the height property currently) but the visual issue it fixes isn't super glaring so I figured we'd just roll with it

File: src/js/sidebar/instructors/search.js (hunk 4 above)
Comment (thomasplevy) [reply src/js/sidebar/instructors/search.js:19]: Yea, when defining properties of an object like this JSDoc requires the main object itself to be defined (with or without a description) prior to the object's properties being defined below: https://jsdoc.app/tags-param.html#parameters-with-properties

elsint throws an error if the property is not defined

**Pull request merged**